### PR TITLE
Dev apis (create_team, join_team, QR code fix)

### PR DESCRIPTION
**Pull request by severma2**

### CODE REVIEW

**Comment by MistrMxyzptlk** [comment]
Tuhle chybu to hází při zapínání uvicornu.

Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/server.py", line 68, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/server.py", line 76, in serve
    config.load()
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/config.py", line 448, in load
    self.loaded_app = import_from_string(self.app)
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/importer.py", line 24, in import_from_string
    raise exc from None
  File "/home/pavel/.local/lib/python3.9/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/lib/python3.9/importlib/init.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/mnt/c/Users/Pavel/Documents/Projekty/shifra/backend/./main.py", line 6, in <module>
    from api.endpoints import user, misc, game, team, cipherPack, hint
ImportError: cannot import name 'cipherPack' from 'api.endpoints' (unknown location)

**Comment by MistrMxyzptlk** [comment]
Internal server error - Once you try to twice with the same user to the same team you get  -> 500 appears
and there is also more complex issue: #98 
And few changes has to be reverted since @SoptikHa2 already soved and pushed them to dev in  #94, #97 .

File: backend/api/endpoints/team.py [hunk 1]
```diff
@@ -25,18 +39,132 @@ def get_team_by_id(team_id: int, response: Response, session_cookie: Optional[st
...
+        return None
+
+    team = Team
+    team.name = team_name
+    team.approved = False
+
+    team_id = teamFuncs.insertTeam(team)
+
+    if team_id is None:
+        # response.status_code = ???
```
Comment (MistrMxyzptlk) [inline backend/api/endpoints/team.py:99]: Použij code 409, není třeba, každý kód využit jen jednou. Chybová odpověď může mít i tělíčko obsahující data o chybě, pokud je to potřeba.

File: backend/api/endpoints/misc.py [hunk 2]
```diff
@@ -24,14 +24,13 @@ def genQR(teamCode: str):
...
+    img = qrcode.make('https://shifra.klubfitpp.cz/team/join/code?code=' + teamCode, image_factory=qrcode.image.svg.SvgImage)
 
     stream = io.BytesIO()
     img.save(stream)
-    stream.seek(0)
 
-    # Save the image into fastapi response stream
-    return StreamingResponse(stream, media_type="image/svg")
+    image = stream.getvalue()
+    return base64.b64encode(image)
```
Comment (MistrMxyzptlk) [inline backend/api/endpoints/misc.py:33]: Tohle jsme se soptíkem vyřešily, prosím roll back. Šlo o špatný content-type

File: backend/api/endpoints/team.py [hunk 3]
```diff
@@ -25,18 +39,132 @@ def get_team_by_id(team_id: int, response: Response, session_cookie: Optional[st
...
+
+    try:
+
+        team_id = teamFuncs.get_id_by_inv_code(inv_code)
+        if team_id is None:
+            response.status_code = 404
+            logger.info(join_team.__name__ + " /api/team/join " + str(response.status_code) + ": team does not exist")
+            return None
+        if teamFuncs.is_full(team_id):
+            # response.status_code = ???
```
Comment (MistrMxyzptlk) [inline backend/api/endpoints/team.py:147]: Použij code 409, není třeba, každý kód využit jen jednou. Chybová odpověď může mít i tělíčko obsahující data o chybě, pokud je to potřeba.

File: backend/api/endpoints/team.py [hunk 4]
```diff
@@ -25,18 +39,132 @@ def get_team_by_id(team_id: int, response: Response, session_cookie: Optional[st
...
+                logger.info(join_team.__name__ + " /api/team/join " + str(response.status_code) + ": user already exists - " + username_cred.username)
+                return None
+            response.status_code = 200
+            joinTeam(team_id, new_user.person_id)
+            return team_id
+
+    except:
+
+        logger.warning(join_team.__name__ + " /api/team/join - something went wrong (Database is down, or user already joined this cipher game)")
+        # response.status_code = ???
```
Comment (MistrMxyzptlk) [inline backend/api/endpoints/team.py:166]: opět 409 conflict

File: backend/db_funcs/cipherGameFuncs.py [hunk 5]
```diff
@@ -85,3 +85,9 @@ def get_all_cipher_games() -> [CipherGame]:
         cur.execute("SELECT * FROM cipher_game cg;")
         result = cur.fetchall()
         return [cipher_game_from_db_row(x) for x in result]
+
+
+# what to return???
```
Comment (MistrMxyzptlk) [inline backend/db_funcs/cipherGameFuncs.py:90]: you shloud return team_id or striped team object

File: backend/base_converter.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+        base = 36
+        while number != 0:
+            remainder = floor(number % base)
+            number = floor(number / base)
+            if remainder > 9:
+                remainder += self.diff
+            result += chr(remainder + ord('0'))
+
+        return result[::-1]
```
Comment (MistrMxyzptlk) [inline backend/base_converter.py:19]: Could be simplified by a lot, but this sould work.

File: backend/routes/person.py [hunk 7]
```diff
@@ -18,4 +18,4 @@ def strip(self):
 
 
 def person_from_db_row(row) -> Person:
-    return Person(person_id=row[0], is_roow=row[1], nickname=row[2], session_cookie=row[3], mail=row[4], password=row[5])
+    return Person(person_id=row[0], is_root=row[1], nickname=row[2], session_cookie=row[3], mail=row[4], password=row[5])
```
Comment (MistrMxyzptlk) [inline backend/routes/person.py:21]: alredy fixed

**Review by MistrMxyzptlk — COMMENTED** [review]
There is comment's, this pull request needs some changes before we can start merging.

File: backend/routes/person.py (hunk 7 above)
Comment (SoptikHa2) [reply backend/routes/person.py:21]: to je ok, mělo by se ztratit po updatu z `dev`

File: backend/api/endpoints/misc.py (hunk 2 above)
Comment (severma2) [reply backend/api/endpoints/misc.py:33]: Ok, po pullu to nechám být

**Review by MistrMxyzptlk — APPROVED** [review]

**Pull request merged**